# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1222667877 -7200
#      Mon Sep 29 07:57:57 2008 +0200
# Branch release-3-0-x
# Node ID 0f3d1dd22905ab50652a1ea198793416b1c175ff
# Parent  5d7aa47a579766b79e594406b502ee6afc62a093
add missing return in Array<T>::delete_elements_2

diff --git a/liboctave/Array.cc b/liboctave/Array.cc
--- a/liboctave/Array.cc
+++ b/liboctave/Array.cc
@@ -1404,16 +1404,18 @@ Array<T>::maybe_delete_elements_2 (idx_v
   else if (! idx_arg.orig_empty ())
     {
       // Reshape to row vector for Matlab compatibility.
 
       n = nr * nc;
       nr = 1;
       nc = n;
     }
+  else
+    return;
 
   idx_arg.sort (true);
 
   if (idx_arg.is_colon_equiv (n, 1))
     {
       if (nr == 1)
         resize_no_fill (1, 0);
       else if (nc == 1)
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-29  Jaroslav Hajek  <highegg@gmail.com>
+
+	* Array.cc (Array<T>::maybe_delete_elements_2(idx_vector&)): Return on
+	empty index vector.
+
 2008-09-29  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Array.cc (Array<T>::maybe_delete_elements_2, Array<T>::maybe_delete_elements): 
 	Ignore null assignments with empty index vectors.
 
 2008-09-19  John W. Eaton  <jwe@octave.org>
 
 	* Array.cc (assign1, assign2, assignN):
